### ISSUE TITLE
RGB colors not restricted to a value between 0 and 255

### ISSUE THREAD

**Issue by ToonTalk** [NetLogo/NetLogo#1491]
http://ccl.northwestern.edu/netlogo/docs/programming.html#colors states "RGB and RGBA lists are made up of three or four integers, respectively, between 0 and 255 if a number is outside that range 255 is repeatedly subtracted until it is in the range."

One can do the following without an error:

set color [-.9 -.9 -.9]
and
set color [255.9 255.9 255.9]

However, contrary to the documentation quoted above

set color [0 0 1000]

signals an error. (Also shouldn't it subtract 256 repeatedly?)

**Comment by mrerrormessage** [comment]
Thanks for reporting. It looks as though the behavior you currently see has been in place at least since NetLogo 5.0. I think the easiest way to resolve this is just to update the documentation to match what's actually implemented. The rules (as far as I can tell) are as follows: that RGB/RGBA lists are

**Comment by ToonTalk** [comment]
Updating the documentation makes sense. But by 'round' you don't seem to mean the NetLogo 'round' reporter where -0.9 rounds to -1 and 255.9 rounds to 256.  'floor' doesn't work for -0.9 either.

**Comment by mrerrormessage** [comment]
> But by 'round' you don't seem to mean the NetLogo 'round' reporter where -0.9 rounds to -1 and 255.9 rounds to 256. 'floor' doesn't work for -0.9 either.

That's correct. Wikipedia lists [several different rounding techniques]. NetLogo's `round` is "Round to nearest" with the tie breaker "round half up". NetLogo's `floor` and `ceiling` are "Round down" and "Round up" respectively. The rounding algorithm used for numbers in an RGB Color list is "Round toward zero."

This algorithm is actually used in many NetLogo primitives which require an integer (`crt` or `item`, for instance) because it's faster than "Round to nearest". As a result, `item -0.9 [1 2 3]` results in `1` instead of an error. Since the vast majority of the primitives that use this technique are defined only for the domain of positive integers, this rounding technique only differs from `floor` for numbers in the range (-1,0). Because this technique is used in several places, it's probably worth documenting it separately (and, of course, linking to it where relevant). The challenge will be about providing the information for advanced users who want to understand how something works without overwhelming novices with information which isn't relevant to them.
